### PR TITLE
support the elasticsearch as search engine

### PR DESCRIPTION
**Pull request by jerrybai2009**
If user want to use elasticsearch as search engine, please specify the search engine in application configuration file and add the elasticsearch index url
search.engine = "elasticsearch"

elasticsearch.dataset.url = $YOUR_DATASET_INDEX_URL
elasticsearch.flow.url = $YOUR_FLOW_INDEX_URL

### CODE REVIEW

**Comment by SunZhaonan** [comment]
Didn't see the logic of pushing data into elasticsearch. Is that a separate procedure?

**Comment by ericsun2** [comment]
@jerrybai2009 please also update the wiki for the ElasticSearch jdbc connection and dataset+flow URL configuration

File: web/public/javascripts/controllers/datasets.js [hunk 1]
```diff
@@ -138,7 +138,7 @@ App.DatasetController = Ember.Controller.extend({
     hasSamples: false,
     isTable: true,
     isJSON: false,
-    ownerTypes: ["", "Producer", "Consumer"],
+    ownerTypes: [],
     userTypes: [{name:"Corporate User", value: "urn:li:corpuser"}, {name:"Group User", value: "urn:li:griduser"}],
```
Comment (ericsun2) [inline web/public/javascripts/controllers/datasets.js]: ownerTypes will contain [Producer, Consumer, Auditor, Delegate]
userTypes will contain [User, Group, Role, Service]

File: web/conf/routes [hunk 2]
```diff
@@ -33,6 +33,8 @@ GET     /lineage/flow/:application/:project/:flow   controllers.Application.flow
 
 GET     /schemaHistory                              controllers.Application.schemaHistory()
 
+GET     /api/v1/owner/types                 controllers.api.v1.Dataset.getDatasetOwnerTypes()
+
 GET     /api/v1/company/entities            controllers.api.v1.User.getAllUserEntities()
```
Comment (ericsun2) [inline web/conf/routes]: /api/v1/party/entities + /api/v1/party/employees + /api/v1/party/groups + /api/v1/party/services + /api/v1/party/roles are better end points than /api/v1/company

**Pull request merged**